Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -2421,144 +2426,152 @@ protected ServerResponse doInBackground(Void... voids) {
+                                            linkCache_.clear();
+                                            prefHelper_.setIdentityID(respJson.getString(Defines.Jsonkey.IdentityID.getKey()));
                                             updateRequestsInQueue = true;
                                         }
                                     }
-                                    
-                                    if (updateRequestsInQueue) {
-                                        updateAllRequestsInQueue();
+                                    if (respJson.has(Defines.Jsonkey.DeviceFingerprintID.getKey())) {
+                                        prefHelper_.setDeviceFingerPrintID(respJson.getString(Defines.Jsonkey.DeviceFingerprintID.getKey()));
+                                        updateRequestsInQueue = true;
                                     }
-                                    
-                                    if (thisReq_ instanceof ServerRequestInitSession) {
-                                        setInitState(SESSION_STATE.INITIALISED);
-                                        thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
-                                        if (!((ServerRequestInitSession) thisReq_).handleBranchViewIfAvailable((serverResponse))) {
-                                            checkForAutoDeepLinkConfiguration();
-                                        }
-                                        // Count down the latch holding getLatestReferringParamsSync
-                                        if (getLatestReferringParamsLatch != null) {
-                                            getLatestReferringParamsLatch.countDown();
-                                        }
-                                        // Count down the latch holding getFirstReferringParamsSync
-                                        if (getFirstReferringParamsLatch != null) {
-                                            getFirstReferringParamsLatch.countDown();
-                                        }
-                                    } else {
-                                        // For setting identity just call only request succeeded
-                                        thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
+                                }
+
+                                if (updateRequestsInQueue) {
+                                    updateAllRequestsInQueue();
+                                }
+
+                                if (thisReq_ instanceof ServerRequestInitSession) {
+                                    setInitState(SESSION_STATE.INITIALISED);
+                                    thisReq_.onRequestSucceeded(serverResponse, branchReferral_);
+                                    if (!((ServerRequestInitSession) thisReq_).handleBranchViewIfAvailable((serverResponse))) {
```

Comment:
I realize you're just shifting/rearranging existing code, but I hadn't seen/noticed some of this before. Do these doubled parens around serverResponse do anything? Is this the same as:

```java
if (!((ServerRequestInitSession)thisReq_).handleBranchViewIfAvailable(serverResponse)) {
```

It would be more readable as

```java
if (thisReq_ instanceof ServeRequestInitSession) {
    ServerRequestInitSession initReq = (ServerRequestInitSession)thisReq_;
    // ...
    if (!initReq.handleBranchViewIfAvailable(serverResponse)) {
```

And for that matter when you have code like this checking `instanceof SomeClass`, it might be time to refactor. But that's obviously out of scope here.

Reply:
There is so much to refactor in this SDK it makes me cringe. In fact I started and doing a big refactor and realized that I'll just make the PR impossible to follow.

Hopefully I can do the reactor before the next release :)